lib/libsmb2.c

```diff
@@ -3319,7 +3317,7 @@ smb2_general_client_request_cb(struct smb2_context *smb2, int status, void *comm
                 smb2_close_context(smb2);
                 return;
         }
-        if (status == SMB2_STATUS_CANCELLED) {
+        if (status == SMB2_STATUS_CANCELLED || status == SMB2_STATUS_SHUTDOWN) {
```

Comment:
CHANGE Note: Important one, the server side will now crash on smb2 context destroy if this change isn't included